src/lib/list/list-option.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterContentInit,
+  Component,
+  ContentChild,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  Optional,
+  QueryList,
+  Renderer2,
+  ViewEncapsulation,
+  ChangeDetectionStrategy,
+  OnDestroy,
+  EventEmitter,
+  Output,
+  ChangeDetectorRef
+} from '@angular/core';
+import {coerceBooleanProperty, MdLine, MdLineSetter} from '../core';
+import {Focusable} from '../core/a11y/focus-key-manager';
+import {MdSelectionList} from './selection-list';
+
+export interface MdSelectionListOptionEvent {
+  option: MdListOption;
+}
+
+const FOCUSED_STYLE: string = 'mat-list-item-focus';
+
+/**
+ * Component for list-options of selection-list. Each list-option can automatically
+ * generate a checkbox and can put current item into the selectionModel of selection-list
+ * if the current item is checked.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'md-list-option, mat-list-option',
+  host: {
+    'role': 'option',
+    'class': 'mat-list-item, mat-list-option',
+    '(focus)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(click)': 'toggle()',
+    'tabindex': '-1',
+    '[attr.aria-selected]': 'selected.toString()',
+    '[attr.aria-disabled]': 'disabled.toString()',
+  },
+  templateUrl: 'list-option.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdListOption implements AfterContentInit, OnDestroy, Focusable {
+  private _lineSetter: MdLineSetter;
+  private _disableRipple: boolean = false;
+  private _selected: boolean = false;
+  /** Whether the checkbox is disabled. */
+  private _disabled: boolean = false;
+  private _value: any;
+
+  /** Whether the option has focus. */
+  _hasFocus: boolean = false;
+
+  /**
+   * Whether the ripple effect on click should be disabled. This applies only to list items that are
+   * part of a nav list. The value of `disableRipple` on the `md-nav-list` overrides this flag.
```

Comment:
List-options are only on selection-lists, right? The comment talks about nav lists, so it looks like this might need to be updated?